src/functionHandlers/notarisePdt/v2/handler.ts

```diff
@@ -51,6 +55,40 @@ export const main: Handler = async (
           );
     }
 
+    try {
+      /* 1.1 Soft Validation with vault data */
+      if (parsedFhirBundle.patient.nricFin) {
+        const personalData = await getPersonalDataFromVault(
+          parsedFhirBundle.patient.nricFin,
+          reference
+        );
+        if (personalData) {
+          const isDobInVault =
+            personalData.dateofbirth === parsedFhirBundle.patient.birthDate;
+          trace(`Is dateofbirth in vault : ${isDobInVault}`);
+          const isGenderInVault =
+            personalData.gender ===
+            parsedFhirBundle.patient.gender?.charAt(0).toUpperCase();
+          trace(`Is gender in vault : ${isGenderInVault}`);
+          const isNameInVault = checkValidPatientName(
+            parsedFhirBundle.patient.fullName,
+            personalData.principalname
+          );
+          trace(`Is name in vault : ${isNameInVault}`);
```

Comment:
- `traceWithRef()`
- Shall we combine all the trace logs into one statement? Easier to debug when searching in Kibana.